Sources/SwiftSMTP/SMTPSocket.swift

```diff
@@ -159,7 +194,13 @@ private extension SMTPSocket {
                 }
             }
         }
-        throw SMTPError.noAuthMethodsOrRequiresTLS(hostname: hostname)
+        if requiresAuth {
```

Comment:
I gave it a test on a server that supports auth, but does not require it. This check prevents sending.

It looks like you'll need to call the correct `SMTPSocket.init` from within `SMTP.send` for those using the new `SMTP.init`.